### PR TITLE
Load type of input, build hidden input, and display keyboard depending on type

### PR DESCRIPTION
**Pull request by pablomarti**

### CODE REVIEW

File: godtools/GeneralExtensions/String.swift [hunk 1]
```diff
@@ -15,6 +15,17 @@ extension String {
         return NSLocalizedString(self, tableName: nil, bundle: Bundle.main, value: "", comment: "")
     }
     
+    func removeBreaklines() -> String {
+        let regex = try! NSRegularExpression(pattern: "\n", options: NSRegularExpression.Options.caseInsensitive)
```
Comment (ryancarlson) [inline godtools/GeneralExtensions/String.swift]: I'm guessing we don't need the class/struct names `NSRegularExpression.Options` here? I'm not sure though.

File: godtools/GeneralExtensions/String.swift [hunk 2]
```diff
@@ -15,6 +15,17 @@ extension String {
...
+        let range = NSMakeRange(0, self.characters.count)
+        return regex.stringByReplacingMatches(in: self, options: [], range: range, withTemplate: "")
+    }
+    
+    func condenseWhitespaces() -> String {
+        let components = self.components(separatedBy: NSCharacterSet.whitespacesAndNewlines)
```
Comment (ryancarlson) [inline godtools/GeneralExtensions/String.swift]: Do we need `NSCharacterSet` here?

File: godtools/Views/TractElements/TractInput.swift [hunk 3]
```diff
@@ -51,17 +51,30 @@ class TractInput: BaseTractElement {
...
         self.textField.backgroundColor = elementProperties.backgroundColor
         self.textField.placeholderTranslationKey = elementProperties.placeholder ?? ""
+        self.textField.text = elementProperties.inputValue()
         self.textField.frame = CGRect(x: 0.0,
                                       y: self.height,
                                       width: self.elementFrame.finalWidth(),
                                       height: elementProperties.height)
         
+        switch elementProperties.type {
+        case .email:
```
Comment (ryancarlson) [inline godtools/Views/TractElements/TractInput.swift]: 👍 !

File: godtools/GeneralExtensions/String.swift (hunk 1 above)
Comment (pablomarti) [inline godtools/GeneralExtensions/String.swift]: Not needed :D

File: godtools/GeneralExtensions/String.swift (hunk 2 above)
Comment (pablomarti) [inline godtools/GeneralExtensions/String.swift]: It is not needed too, just making the changes, thanks!

**Pull request merged**